tests/scripts/restore-mon-quorum-using-osds.sh

```diff
@@ -0,0 +1,319 @@
+    rm -rf "${TMP_MON_STORE}"
+    mkdir -p "${TMP_MON_STORE}"
+
+    MONS=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-mon -o json | jq -r '.items[].metadata.name')
+    DESIRED_MON_COUNT=$(echo ${MONS} | wc -w)
+    if [ ${DESIRED_MON_COUNT} = 0 ]; then
+        echo "the desired number of MON should be >= 1: ${DESIRED_OSD_COUNT}" >&2
+        exit 1
+    fi
+    OSDS=$(kubectl get deployment -n "${NAMESPACE}" -l app=rook-ceph-osd -o json | jq -r '.items[].metadata.name')
+    DESIRED_OSD_COUNT=$(echo ${OSDS} | wc -w)
+    if [ $DESIRED_OSD_COUNT = 0 ]; then
+        echo "the desired number of OSD should be >= 1: ${DESIRED_OSD_COUNT}" >&2
+        exit 1
+    fi
+
+    # MON_IDS should be sorted by IP addresses due to the limitation of ceph-monstore-tool. See the following Ceph's issue for more detail.
+    # https://tracker.ceph.com/issues/49158
+    MON_IDS=$(kubectl get svc -n "${NAMESPACE}" -l app=rook-ceph-mon -o json | jq -r '.items[] | [.metadata.name, .spec.clusterIP] | @sh' | tr -d \' |  sort -t . -k 2,2n -k 3,3n -k 4,4n | cut -d' ' -f1 |  cut -d'-' -f4)
+    OSD_IDS=$(kubectl get -n "${NAMESPACE}" po -l app=rook-ceph-osd -o json | jq -r '.items[].metadata.labels["osd"]')
+
+    TOOL=$(kubectl get -n "${NAMESPACE}" po -l app=rook-ceph-tools -o json | jq -r '.items[0].metadata.name')
+}
+
+RESTORE=0
+while getopts hrn:b:t: OPT
+do
+  case "${OPT}" in
+    h)
+        usage 0
+        ;;
+    n)
+        NAMESPACE="${OPTARG}"
+        ;;
+    b)
+        BACKUP_DIR="${OPTARG}"
+        ;;
+    t)
+        TMP_DIR="${OPTARG}"
+        ;;
+    r)
+        RESTORE=1
+        ;;
+    *)
+        echo "invalid option: ${OPT}"
+        usage 1
+        ;;
+  esac
+done
+
+shift $((OPTIND - 1))
+
+if [ "${RESTORE}" = 1 ]; then
+  if [ -z "${NAMESPACE}" -o -z "${BACKUP_DIR}" ]; then
+    usage 1
+  fi
+  init
+  restore
+  exit 0
+fi
+
+if [ -z "${NAMESPACE}" -o -z "${BACKUP_DIR}" -o -z "${TMP_DIR}" ]; then
+    usage 1
+fi
+
+init
+
+echo '[STEP1] taking backup'
+
+for mon in ${MONS}; do
+    kubectl -n "${NAMESPACE}" get deployment "${mon}" -o yaml > "${BACKUP_DIR}"/"${mon}".yaml
+done
+
+for osd in ${OSDS}; do
+    kubectl -n "${NAMESPACE}" get deployment "${osd}" -o yaml > "${BACKUP_DIR}"/"${osd}".yaml
+done
+
+kubectl -n "${NAMESPACE}" get deployment "${TOOL_DEPLOYMENT}" -o yaml > "${BACKUP_DIR}"/"${TOOL_DEPLOYMENT}".yaml
+
+echo '[STEP1] done'
+
+echo '[STEP2] suspending operator pod'
+
+kubectl -n "${NAMESPACE}" scale deployment rook-ceph-operator --replicas=0
+wait_pods rook-ceph-operator 0
+
+echo '[STEP2] done'
+
+echo '[STEP3] suspending mon pods and osd pods'
+
+for mon in ${MONS}; do
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" -p '{"spec": {"template": {"spec": {"containers": [{"name": "mon", "command": ["sleep", "infinity"], "args": []}]}}}}'
+    kubectl -n "${NAMESPACE}" patch deployment "${mon}" --type=json --patch='[{"op":"remove", "path":"/spec/template/spec/containers/0/livenessProbe"}]'
```

Comment:
Always return an error for me on rook v1.2.7, I think this kind of error must be ignored.

Reply:
The target of this script is only master branch.